lib/partition.coffee

```diff
@@ -114,10 +116,35 @@ exports.getPartitionOffset = (partition) ->
 	if not _.isNumber(partition.firstLBA)
 		throw new errors.ResinInvalidOption('firstLBA', partition.firstLBA, 'not a number')
 
-	return partition.firstLBA * 512
+	return partition.firstLBA * SECTOR_SIZE
 
 ###*
-# @summary Get a partition position
+# @summary Get the partition size in bytes
+# @protected
+# @function
+#
+# @param {Object} partition - partition
+# @returns {Number} partition size
+#
+# @example
+# size = partition.getPartitionSize(myPartition)
+###
+exports.getPartitionSize = (partition) ->
+
+	if not partition?
+		throw new errors.ResinMissingParameter('partition')
+
+	if not partition.sectors?
+		throw new errors.ResinMissingOption('sectors')
+
+	if not _.isNumber(partition.sectors)
+		throw new errors.ResinInvalidOption('sectors', partition.sectors, 'not a number')
+
+	# TODO: Why does the magic number 5 works here?
+	return partition.sectors * SECTOR_SIZE / 5
```

Comment:
@abresas Notice I'm interacting with the partition trough fatfs, so I suspect that module is taking care of adding zeroes when writing to the partition and cutting off the \0s when reading from it.

The only modification to node-mbr output happens within this same file, below in the `getPartitionFromDefiniton()` function, where I sum the primary partition's `firstLBA` to the logical partition's `firstLBA`.